test/tasks/before-suite.yaml

```diff
@@ -0,0 +1,107 @@
+# ginkgo-suite is a generic task that executes ginkgo on the given repository and ginkgo suite.
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: before-suite
+  namespace: karpenter-tests
+spec:
+  description: |
+    Execute a karpenter test suite
+  params:
+    - name: cluster-name
+      description: Name of the cluster under test.
+    - name: git-ref
+      description: Git commit, tag, or branch to check out. Requires a corresponding Karpenter snapshot release.
+  results:
+    - name: account-id
+    - name: region
+  workspaces:
+    - name: ws
+  steps:
+  - name: detect-environment
+    image: 767520670908.dkr.ecr.us-west-2.amazonaws.com/karpenter-test-infrastructure:v4
+    workingDir: $(workspaces.ws.path)
+    script: |
+      TOKEN=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
+      curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/identity-credentials/ec2/info | grep "AccountId" | cut -f4 -d"\"" > $(results.account-id.path)
```

Comment:
https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/instance-identity-documents.html
Can we use jq instead of grep here for robustness?